### ISSUE TITLE
ConnectTimeoutException during redirect

### ISSUE THREAD

**Issue by khauser** [rancher/rancher#7025]
**Rancher Version:**
1.2.0
**Docker Version:**
1.12.4
**OS and where are the hosts located? (cloud, bare metal, etc):**
Ubuntu
**Setup Details: (single node rancher vs. HA rancher, internal DB vs. external DB)**
single node rancher
**Environment Type: (Cattle/Kubernetes/Swarm/Mesos)**
Cattle I think

After upgrade from Rancher 1.1.3 I can not come around this failure.

```
13.12.2016 11:08:322016-12-13 10:08:32.192  WARN 1 --- [-nio-443-exec-6] o.s.c.n.z.filters.post.SendErrorFilter   : Error during filtering
13.12.2016 11:08:32
13.12.2016 11:08:32org.apache.http.conn.ConnectTimeoutException: Connect to <my_domain>:443 [<my_domain>/139.59.212.123] failed: connect timed out
13.12.2016 11:08:32	at org.apache.http.impl.conn.DefaultHttpClientConnectionOperator.connect(DefaultHttpClientConnectionOperator.java:150) ~[httpclient-4.5.2.jar!/:4.5.2]
13.12.2016 11:08:32	at org.apache.http.impl.conn.PoolingHttpClientConnectionManager.connect(PoolingHttpClientConnectionManager.java:353) ~[httpclient-4.5.2.jar!/:4.5.2]
13.12.2016 11:08:32	at org.apache.http.impl.execchain.MainClientExec.establishRoute(MainClientExec.java:380) ~[httpclient-4.5.2.jar!/:4.5.2]
13.12.2016 11:08:32	at org.apache.http.impl.execchain.MainClientExec.execute(MainClientExec.java:236) ~[httpclient-4.5.2.jar!/:4.5.2]
13.12.2016 11:08:32	at org.apache.http.impl.execchain.ProtocolExec.execute(ProtocolExec.java:184) ~[httpclient-4.5.2.jar!/:4.5.2]
13.12.2016 11:08:32	at org.apache.http.impl.execchain.RetryExec.execute(RetryExec.java:88) ~[httpclient-4.5.2.jar!/:4.5.2]
13.12.2016 11:08:32	at org.apache.http.impl.execchain.RedirectExec.execute(RedirectExec.java:110) ~[httpclient-4.5.2.jar!/:4.5.2]
13.12.2016 11:08:32	at org.apache.http.impl.client.InternalHttpClient.doExecute(InternalHttpClient.java:184) ~[httpclient-4.5.2.jar!/:4.5.2]
13.12.2016 11:08:32	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:117) ~[httpclient-4.5.2.jar!/:4.5.2]
13.12.2016 11:08:32	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:55) ~[httpclient-4.5.2.jar!/:4.5.2]
13.12.2016 11:08:32	at org.springframework.cloud.netflix.zuul.filters.route.SimpleHostRoutingFilter.forwardRequest(SimpleHostRoutingFilter.java:358) ~[spring-cloud-netflix-core-1.2.3.RELEASE.jar!/:1.2.3.RELEASE]
13.12.2016 11:08:32	at org.springframework.cloud.netflix.zuul.filters.route.SimpleHostRoutingFilter.forward(SimpleHostRoutingFilter.java:284) ~[spring-cloud-netflix-core-1.2.3.RELEASE.jar!/:1.2.3.RELEASE]
13.12.2016 11:08:32	at org.springframework.cloud.netflix.zuul.filters.route.SimpleHostRoutingFilter.run(SimpleHostRoutingFilter.java:180) ~[spring-cloud-netflix-core-1.2.3.RELEASE.jar!/:1.2.3.RELEASE]
13.12.2016 11:08:32	at com.netflix.zuul.ZuulFilter.runFilter(ZuulFilter.java:112) ~[zuul-core-1.3.0.jar!/:1.3.0]
13.12.2016 11:08:32	at com.netflix.zuul.FilterProcessor.processZuulFilter(FilterProcessor.java:193) ~[zuul-core-1.3.0.jar!/:1.3.0]
13.12.2016 11:08:32	at com.netflix.zuul.FilterProcessor.runFilters(FilterProcessor.java:157) ~[zuul-core-1.3.0.jar!/:1.3.0]
13.12.2016 11:08:32	at com.netflix.zuul.FilterProcessor.route(FilterProcessor.java:118) ~[zuul-core-1.3.0.jar!/:1.3.0]
13.12.2016 11:08:32	at com.netflix.zuul.ZuulRunner.route(ZuulRunner.java:96) ~[zuul-core-1.3.0.jar!/:1.3.0]
13.12.2016 11:08:32	at com.netflix.zuul.http.ZuulServlet.route(ZuulServlet.java:116) ~[zuul-core-1.3.0.jar!/:1.3.0]
13.12.2016 11:08:32	at com.netflix.zuul.http.ZuulServlet.service(ZuulServlet.java:81) ~[zuul-core-1.3.0.jar!/:1.3.0]
13.12.2016 11:08:32	at org.springframework.web.servlet.mvc.ServletWrappingController.handleRequestInternal(ServletWrappingController.java:157) [spring-webmvc-4.3.4.RELEASE.jar!/:4.3.4.RELEASE]
13.12.2016 11:08:32	at org.springframework.cloud.netflix.zuul.web.ZuulController.handleRequest(ZuulController.java:44) [spring-cloud-netflix-core-1.2.3.RELEASE.jar!/:1.2.3.RELEASE]
13.12.2016 11:08:32	at org.springframework.web.servlet.mvc.SimpleControllerHandlerAdapter.handle(SimpleControllerHandlerAdapter.java:50) [spring-webmvc-4.3.4.RELEASE.jar!/:4.3.4.RELEASE]
13.12.2016 11:08:32	at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:963) [spring-webmvc-4.3.4.RELEASE.jar!/:4.3.4.RELEASE]
13.12.2016 11:08:32	at org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:897) [spring-webmvc-4.3.4.RELEASE.jar!/:4.3.4.RELEASE]
13.12.2016 11:08:32	at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:970) [spring-webmvc-4.3.4.RELEASE.jar!/:4.3.4.RELEASE]
13.12.2016 11:08:32	at org.springframework.web.servlet.FrameworkServlet.doGet(FrameworkServlet.java:861) [spring-webmvc-4.3.4.RELEASE.jar!/:4.3.4.RELEASE]
13.12.2016 11:08:32	at javax.servlet.http.HttpServlet.service(HttpServlet.java:622) [tomcat-embed-core-8.5.6.jar!/:8.5.6]
13.12.2016 11:08:32	at org.springframework.web.servlet.FrameworkServlet.service(FrameworkServlet.java:846) [spring-webmvc-4.3.4.RELEASE.jar!/:4.3.4.RELEASE]
13.12.2016 11:08:32	at javax.servlet.http.HttpServlet.service(HttpServlet.java:729) [tomcat-embed-core-8.5.6.jar!/:8.5.6]
13.12.2016 11:08:32	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:230) [tomcat-embed-core-8.5.6.jar!/:8.5.6]
13.12.2016 11:08:32	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:165) [tomcat-embed-core-8.5.6.jar!/:8.5.6]
13.12.2016 11:08:32	at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:52) [tomcat-embed-websocket-8.5.6.jar!/:8.5.6]
13.12.2016 11:08:32	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:192) [tomcat-embed-core-8.5.6.jar!/:8.5.6]
13.12.2016 11:08:32	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:165) [tomcat-embed-core-8.5.6.jar!/:8.5.6]
13.12.2016 11:08:32	at org.springframework.boot.web.filter.ApplicationContextHeaderFilter.doFilterInternal(ApplicationContextHeaderFilter.java:55) [spring-boot-1.4.2.RELEASE.jar!/:1.4.2.RELEASE]
13.12.2016 11:08:32	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107) [spring-web-4.3.4.RELEASE.jar!/:4.3.4.RELEASE]
13.12.2016 11:08:32	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:192) [tomcat-embed-core-8.5.6.jar!/:8.5.6]
13.12.2016 11:08:32	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:165) [tomcat-embed-core-8.5.6.jar!/:8.5.6]
13.12.2016 11:08:32	at org.springframework.boot.actuate.trace.WebRequestTraceFilter.doFilterInternal(WebRequestTraceFilter.java:105) [spring-boot-actuator-1.4.2.RELEASE.jar!/:1.4.2.RELEASE]
13.12.2016 11:08:32	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107) [spring-web-4.3.4.RELEASE.jar!/:4.3.4.RELEASE]
13.12.2016 11:08:32	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:192) [tomcat-embed-core-8.5.6.jar!/:8.5.6]
13.12.2016 11:08:32	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:165) [tomcat-embed-core-8.5.6.jar!/:8.5.6]
13.12.2016 11:08:32	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:317) [spring-security-web-4.1.3.RELEASE.jar!/:4.1.3.RELEASE]
13.12.2016 11:08:32	at org.springframework.security.web.access.intercept.FilterSecurityInterceptor.invoke(FilterSecurityInterceptor.java:127) [spring-security-web-4.1.3.RELEASE.jar!/:4.1.3.RELEASE]
13.12.2016 11:08:32	at org.springframework.security.web.access.intercept.FilterSecurityInterceptor.doFilter(FilterSecurityInterceptor.java:91) [spring-security-web-4.1.3.RELEASE.jar!/:4.1.3.RELEASE]
13.12.2016 11:08:32	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331) [spring-security-web-4.1.3.RELEASE.jar!/:4.1.3.RELEASE]
13.12.2016 11:08:32	at org.springframework.security.web.access.ExceptionTranslationFilter.doFilter(ExceptionTranslationFilter.java:115) [spring-security-web-4.1.3.RELEASE.jar!/:4.1.3.RELEASE]
13.12.2016 11:08:32	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331) [spring-security-web-4.1.3.RELEASE.jar!/:4.1.3.RELEASE]
13.12.2016 11:08:32	at org.springframework.security.web.session.SessionManagementFilter.doFilter(SessionManagementFilter.java:137) [spring-security-web-4.1.3.RELEASE.jar!/:4.1.3.RELEASE]
13.12.2016 11:08:32	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331) [spring-security-web-4.1.3.RELEASE.jar!/:4.1.3.RELEASE]
13.12.2016 11:08:32	at org.springframework.security.web.authentication.AnonymousAuthenticationFilter.doFilter(AnonymousAuthenticationFilter.java:111) [spring-security-web-4.1.3.RELEASE.jar!/:4.1.3.RELEASE]
13.12.2016 11:08:32	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331) [spring-security-web-4.1.3.RELEASE.jar!/:4.1.3.RELEASE]
13.12.2016 11:08:32	at org.springframework.security.web.servletapi.SecurityContextHolderAwareRequestFilter.doFilter(SecurityContextHolderAwareRequestFilter.java:169) [spring-security-web-4.1.3.RELEASE.jar!/:4.1.3.RELEASE]
13.12.2016 11:08:32	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331) [spring-security-web-4.1.3.RELEASE.jar!/:4.1.3.RELEASE]
13.12.2016 11:08:32	at org.springframework.security.web.savedrequest.RequestCacheAwareFilter.doFilter(RequestCacheAwareFilter.java:63) [spring-security-web-4.1.3.RELEASE.jar!/:4.1.3.RELEASE]
13.12.2016 11:08:32	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331) [spring-security-web-4.1.3.RELEASE.jar!/:4.1.3.RELEASE]
13.12.2016 11:08:32	at org.springframework.security.web.authentication.AbstractAuthenticationProcessingFilter.doFilter(AbstractAuthenticationProcessingFilter.java:200) [spring-security-web-4.1.3.RELEASE.jar!/:4.1.3.RELEASE]
13.12.2016 11:08:32	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331) [spring-security-web-4.1.3.RELEASE.jar!/:4.1.3.RELEASE]
13.12.2016 11:08:32	at org.springframework.security.web.authentication.logout.LogoutFilter.doFilter(LogoutFilter.java:121) [spring-security-web-4.1.3.RELEASE.jar!/:4.1.3.RELEASE]
13.12.2016 11:08:32	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331) [spring-security-web-4.1.3.RELEASE.jar!/:4.1.3.RELEASE]
13.12.2016 11:08:32	at org.springframework.security.web.header.HeaderWriterFilter.doFilterInternal(HeaderWriterFilter.java:66) [spring-security-web-4.1.3.RELEASE.jar!/:4.1.3.RELEASE]
13.12.2016 11:08:32	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107) [spring-web-4.3.4.RELEASE.jar!/:4.3.4.RELEASE]
13.12.2016 11:08:32	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331) [spring-security-web-4.1.3.RELEASE.jar!/:4.1.3.RELEASE]
13.12.2016 11:08:32	at org.springframework.security.web.context.SecurityContextPersistenceFilter.doFilter(SecurityContextPersistenceFilter.java:105) [spring-security-web-4.1.3.RELEASE.jar!/:4.1.3.RELEASE]
13.12.2016 11:08:32	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331) [spring-security-web-4.1.3.RELEASE.jar!/:4.1.3.RELEASE]
13.12.2016 11:08:32	at org.springframework.security.web.context.request.async.WebAsyncManagerIntegrationFilter.doFilterInternal(WebAsyncManagerIntegrationFilter.java:56) [spring-security-web-4.1.3.RELEASE.jar!/:4.1.3.RELEASE]
13.12.2016 11:08:32	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107) [spring-web-4.3.4.RELEASE.jar!/:4.3.4.RELEASE]
13.12.2016 11:08:32	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331) [spring-security-web-4.1.3.RELEASE.jar!/:4.1.3.RELEASE]
13.12.2016 11:08:32	at org.springframework.security.web.access.channel.ChannelProcessingFilter.doFilter(ChannelProcessingFilter.java:157) [spring-security-web-4.1.3.RELEASE.jar!/:4.1.3.RELEASE]
13.12.2016 11:08:32	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331) [spring-security-web-4.1.3.RELEASE.jar!/:4.1.3.RELEASE]
13.12.2016 11:08:32	at org.springframework.security.web.FilterChainProxy.doFilterInternal(FilterChainProxy.java:214) [spring-security-web-4.1.3.RELEASE.jar!/:4.1.3.RELEASE]
13.12.2016 11:08:32	at org.springframework.security.web.FilterChainProxy.doFilter(FilterChainProxy.java:177) [spring-security-web-4.1.3.RELEASE.jar!/:4.1.3.RELEASE]
13.12.2016 11:08:32	at org.springframework.web.filter.DelegatingFilterProxy.invokeDelegate(DelegatingFilterProxy.java:346) [spring-web-4.3.4.RELEASE.jar!/:4.3.4.RELEASE]
13.12.2016 11:08:32	at org.springframework.web.filter.DelegatingFilterProxy.doFilter(DelegatingFilterProxy.java:262) [spring-web-4.3.4.RELEASE.jar!/:4.3.4.RELEASE]
13.12.2016 11:08:32	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:192) [tomcat-embed-core-8.5.6.jar!/:8.5.6]
13.12.2016 11:08:32	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:165) [tomcat-embed-core-8.5.6.jar!/:8.5.6]
13.12.2016 11:08:32	at org.springframework.web.filter.RequestContextFilter.doFilterInternal(RequestContextFilter.java:99) [spring-web-4.3.4.RELEASE.jar!/:4.3.4.RELEASE]
13.12.2016 11:08:32	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107) [spring-web-4.3.4.RELEASE.jar!/:4.3.4.RELEASE]
13.12.2016 11:08:32	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:192) [tomcat-embed-core-8.5.6.jar!/:8.5.6]
13.12.2016 11:08:32	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:165) [tomcat-embed-core-8.5.6.jar!/:8.5.6]
13.12.2016 11:08:32	at org.springframework.security.oauth2.client.filter.OAuth2ClientContextFilter.doFilter(OAuth2ClientContextFilter.java:60) [spring-security-oauth2-2.0.12.RELEASE.jar!/:na]
13.12.2016 11:08:32	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:192) [tomcat-embed-core-8.5.6.jar!/:8.5.6]
13.12.2016 11:08:32	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:165) [tomcat-embed-core-8.5.6.jar!/:8.5.6]
13.12.2016 11:08:32	at org.springframework.web.filter.HttpPutFormContentFilter.doFilterInternal(HttpPutFormContentFilter.java:89) [spring-web-4.3.4.RELEASE.jar!/:4.3.4.RELEASE]
13.12.2016 11:08:32	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107) [spring-web-4.3.4.RELEASE.jar!/:4.3.4.RELEASE]
13.12.2016 11:08:32	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:192) [tomcat-embed-core-8.5.6.jar!/:8.5.6]
13.12.2016 11:08:32	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:165) [tomcat-embed-core-8.5.6.jar!/:8.5.6]
13.12.2016 11:08:32	at org.springframework.web.filter.HiddenHttpMethodFilter.doFilterInternal(HiddenHttpMethodFilter.java:77) [spring-web-4.3.4.RELEASE.jar!/:4.3.4.RELEASE]
13.12.2016 11:08:32	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107) [spring-web-4.3.4.RELEASE.jar!/:4.3.4.RELEASE]
13.12.2016 11:08:32	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:192) [tomcat-embed-core-8.5.6.jar!/:8.5.6]
13.12.2016 11:08:32	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:165) [tomcat-embed-core-8.5.6.jar!/:8.5.6]
13.12.2016 11:08:32	at org.springframework.session.web.http.SessionRepositoryFilter.doFilterInternal(SessionRepositoryFilter.java:164) [spring-session-1.2.1.RELEASE.jar!/:na]
13.12.2016 11:08:32	at org.springframework.session.web.http.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:80) [spring-session-1.2.1.RELEASE.jar!/:na]
13.12.2016 11:08:32	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:192) [tomcat-embed-core-8.5.6.jar!/:8.5.6]
13.12.2016 11:08:32	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:165) [tomcat-embed-core-8.5.6.jar!/:8.5.6]
13.12.2016 11:08:32	at org.springframework.web.filter.CharacterEncodingFilter.doFilterInternal(CharacterEncodingFilter.java:197) [spring-web-4.3.4.RELEASE.jar!/:4.3.4.RELEASE]
13.12.2016 11:08:32	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107) [spring-web-4.3.4.RELEASE.jar!/:4.3.4.RELEASE]
13.12.2016 11:08:32	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:192) [tomcat-embed-core-8.5.6.jar!/:8.5.6]
13.12.2016 11:08:32	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:165) [tomcat-embed-core-8.5.6.jar!/:8.5.6]
13.12.2016 11:08:32	at org.springframework.boot.actuate.autoconfigure.MetricsFilter.doFilterInternal(MetricsFilter.java:107) [spring-boot-actuator-1.4.2.RELEASE.jar!/:1.4.2.RELEASE]
13.12.2016 11:08:32	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107) [spring-web-4.3.4.RELEASE.jar!/:4.3.4.RELEASE]
13.12.2016 11:08:32	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:192) [tomcat-embed-core-8.5.6.jar!/:8.5.6]
13.12.2016 11:08:32	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:165) [tomcat-embed-core-8.5.6.jar!/:8.5.6]
13.12.2016 11:08:32	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:198) [tomcat-embed-core-8.5.6.jar!/:8.5.6]
13.12.2016 11:08:32	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:108) [tomcat-embed-core-8.5.6.jar!/:8.5.6]
13.12.2016 11:08:32	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:472) [tomcat-embed-core-8.5.6.jar!/:8.5.6]
13.12.2016 11:08:32	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:140) [tomcat-embed-core-8.5.6.jar!/:8.5.6]
13.12.2016 11:08:32	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:79) [tomcat-embed-core-8.5.6.jar!/:8.5.6]
13.12.2016 11:08:32	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:87) [tomcat-embed-core-8.5.6.jar!/:8.5.6]
13.12.2016 11:08:32	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:349) [tomcat-embed-core-8.5.6.jar!/:8.5.6]
13.12.2016 11:08:32	at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:784) [tomcat-embed-core-8.5.6.jar!/:8.5.6]
13.12.2016 11:08:32	at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:66) [tomcat-embed-core-8.5.6.jar!/:8.5.6]
13.12.2016 11:08:32	at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:802) [tomcat-embed-core-8.5.6.jar!/:8.5.6]
13.12.2016 11:08:32	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1410) [tomcat-embed-core-8.5.6.jar!/:8.5.6]
13.12.2016 11:08:32	at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49) [tomcat-embed-core-8.5.6.jar!/:8.5.6]
13.12.2016 11:08:32	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [na:1.8.0_92]
13.12.2016 11:08:32	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [na:1.8.0_92]
13.12.2016 11:08:32	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61) [tomcat-embed-core-8.5.6.jar!/:8.5.6]
13.12.2016 11:08:32	at java.lang.Thread.run(Thread.java:745) [na:1.8.0_92]
13.12.2016 11:08:32Caused by: java.net.SocketTimeoutException: connect timed out
13.12.2016 11:08:32	at java.net.PlainSocketImpl.socketConnect(Native Method) ~[na:1.8.0_92]
13.12.2016 11:08:32	at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:350) ~[na:1.8.0_92]
13.12.2016 11:08:32	at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:206) ~[na:1.8.0_92]
13.12.2016 11:08:32	at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:188) ~[na:1.8.0_92]
13.12.2016 11:08:32	at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392) ~[na:1.8.0_92]
13.12.2016 11:08:32	at java.net.Socket.connect(Socket.java:589) ~[na:1.8.0_92]
13.12.2016 11:08:32	at org.apache.http.conn.ssl.SSLConnectionSocketFactory.connectSocket(SSLConnectionSocketFactory.java:337) ~[httpclient-4.5.2.jar!/:4.5.2]
13.12.2016 11:08:32	at org.apache.http.impl.conn.DefaultHttpClientConnectionOperator.connect(DefaultHttpClientConnectionOperator.java:141) ~[httpclient-4.5.2.jar!/:4.5.2]
13.12.2016 11:08:32	... 119 common frames omitted
13.12.2016 11:08:32
```

Here is also my **docker-compose.yml** which worked with 1.1.3:

```
version: '2'
services:
  hbci:
    image: <image_hidden>
    stdin_open: true
    tty: true
    links:
    - discovery:discovery
    - backend:backend
    labels:
      io.rancher.scheduler.affinity:host_label: system=frontend
      io.rancher.container.hostname_override: container_name
      io.rancher.container.pull_image: always
  eventstore:
    mem_limit: 104857600
    image: <image_hidden>
    links:
    - discovery:discovery
    - mysql:mysql
    - rabbitmq:rabbitmq
    - mongodb:mongodb
    labels:
      io.rancher.scheduler.affinity:host_label: system=utils
      io.rancher.container.hostname_override: container_name
      io.rancher.container.pull_image: always
  edge:
    mem_limit: 314572800
    image: <image_hidden>
    volumes:
    - /ssl:/ssl
    links:
    - frontend:frontend
    - discovery:discovery
    - redis:redis
    - backend:backend
    - authserver:authserver
    - backend:backend
    - frontend:frontend
    - authserver:authserver
    - discovery:discovery
    - redis:redis
    ports:
    - 443:443/tcp
    - 80:80/tcp
    labels:
      io.rancher.scheduler.affinity:host_label: system=frontend
      io.rancher.container.hostname_override: container_name
      io.rancher.container.pull_image: always
  discovery:
    mem_limit: 157286400
    image: <image_hidden>
    ports:
    - 8761:8761/tcp
    labels:
      io.rancher.scheduler.affinity:host_label: system=utils
      io.rancher.container.hostname_override: container_name
      io.rancher.container.pull_image: always
  authserver:
    mem_limit: 104857600
    image: <image_hidden>
    volumes:
    - /ssl:/ssl
    links:
    - eventstore:eventstore
    - discovery:discovery
    - redis:redis
    - mysql:mysql
    labels:
      io.rancher.scheduler.affinity:host_label: system=frontend
      io.rancher.container.hostname_override: container_name
      io.rancher.container.pull_image: always
  userservice:
    mem_limit: 209715200
    image: <image_hidden>
    links:
    - discovery:discovery
    - mysql:mysql
    - rabbitmq:rabbitmq
    labels:
      io.rancher.scheduler.affinity:host_label: system=utils
      io.rancher.container.hostname_override: container_name
      io.rancher.container.pull_image: always
  mysql:
    image: mysql:5.7
    environment:
      MYSQL_ALLOW_EMPTY_PASSWORD: 'yes'
      MYSQL_USER: root
    volumes:
    - /mysql/conf.d:/etc/mysql/conf.d
    - /mysql/data:/var/lib/mysql
    labels:
      io.rancher.scheduler.affinity:host_label: system=database
      io.rancher.container.hostname_override: container_name
  backend:
    mem_limit: 524288000
    image: <image_hidden>
    links:
    - discovery:discovery
    - mysql:mysql
    - rabbitmq:rabbitmq
    labels:
      io.rancher.scheduler.affinity:host_label: system=utils
      io.rancher.container.hostname_override: container_name
      io.rancher.container.pull_image: always
  rabbitmq:
    image: rabbitmq:3.5.1
    labels:
      io.rancher.scheduler.affinity:host_label: system=database
      io.rancher.container.hostname_override: container_name
  mongodb:
    image: mongo:3.2.11
    stdin_open: true
    volumes:
    - /mongo/data:/data/db
    tty: true
    command:
    - --smallfiles
    labels:
      io.rancher.scheduler.affinity:host_label: system=database
      io.rancher.container.hostname_override: container_name
      io.rancher.container.pull_image: always
  redis:
    image: redis:3.2.1-alpine
    labels:
      io.rancher.scheduler.affinity:host_label: system=database
      io.rancher.container.hostname_override: container_name
  frontend:
    mem_limit: 524288000
    image: <image_hidden>
    volumes:
    - /ssl:/ssl
    links:
    - userservice:userservice
    - discovery:discovery
    - redis:redis
    - backend:backend
    labels:
      io.rancher.scheduler.affinity:host_label: system=frontend
      io.rancher.container.hostname_override: container_name
      io.rancher.container.pull_image: always
```

The error comes doing a login at the api-gateway/edge service. After authentication a redirect to the domain should be done. But this fails.

Thanks for your help!
Karsten

**Comment by khauser** [comment]
I added file to a base project which should be able to reproduce this issue. You can find it here:
https://github.com/khauser/microservices4vaadin/blob/master/rancher-docker-compose.yml

Maybe somebody finds time to look init it.

**Comment by khauser** [comment]
Still an issue in 1.2.1

**Comment by khauser** [comment]
Fixed now with rancher/network-manager:v0.2.19 i think. Great work, and happy christmas ;)!
